mne/viz/ica.py

```diff
@@ -130,6 +130,264 @@ def plot_ica_sources(ica, inst, picks=None, exclude=None, start=None,
+            erp_std = [
+                [np.sqrt((d[d < 0] ** 2).mean(axis=0)) for d in diffs.T],
+                [np.sqrt((d[d > 0] ** 2).mean(axis=0)) for d in diffs.T]]
+            erp_std = np.array(erp_std) * num_std
+
+        # epoch variance
+        epoch_var = np.var(ica_data[idx], axis=1)
+
+        # plotting
+        # --------
+        # component topomap
+        _plot_ica_topomap(ica, pick, show=False, axis=axes[0], **topomap_args)
+
+        # image and erp
+        plot_epochs_image(epochs_src, picks=pick, axes=axes[1:3],
+                          colorbar=False, show=False, **image_args)
+
+        # spectrum
+        axes[3].plot(freqs, psds_mean, color='k')
+        if plot_std:
+            axes[3].fill_between(freqs, psds_mean - spectrum_std[0],
+                                 psds_mean + spectrum_std[1],
+                                 color='k', alpha=.15)
+        if plot_lowpass_edge:
+            axes[3].axvline(inst.info['lowpass'], lw=2, linestyle='--',
+                            color='k', alpha=0.15)
+
+        # epoch variance
+        axes[4].scatter(range(len(epoch_var)), epoch_var, alpha=0.5,
+                        facecolor=[0, 0, 0], lw=0)
+
+        # aesthetics
+        # ----------
+        axes[0].set_title('IC #{:0>3}'.format(str(pick)))
+
+        set_title_and_labels(axes[1], 'epochs image and ERP/ERF', [], 'Epochs')
+
+        # erp
+        set_title_and_labels(axes[2], [], 'time', 'AU')
+        # line color and std
+        axes[2].lines[0].set_color('k')
+        if plot_std:
+            erp_xdata = axes[2].lines[0].get_data()[0]
+            axes[2].fill_between(erp_xdata, erp - erp_std[0],
+                                 erp + erp_std[1], color='k', alpha=.15)
+            axes[2].autoscale(enable=True, axis='y')
+            axes[2].axis('auto')
+            axes[2].set_xlim(erp_xdata[[0, -1]])
+        # remove half of yticks if more than 5
+        yt = axes[2].get_yticks()
+        if len(yt) > 5:
+            yt = yt[::2]
+            axes[2].set_yticks(yt)
+
+        if not plot_line_at_zero:
+            xlims = [1e3 * inst.times[0], 1e3 * inst.times[-1]]
+            for k, ax in enumerate(axes[1:3]):
+                ax.lines[k].remove()
+                ax.set_xlim(xlims)
+
+        # remove xticks - erp plot shows xticks for both image and erp plot
+        axes[1].set_xticks([])
+        yt = axes[1].get_yticks()
+        axes[1].set_yticks(yt[1:])
+        axes[1].set_ylim([-0.5, ica_data.shape[1] + 0.5])
+
+        # spectrum
+        ylabel = 'dB' if dB else 'power'
+        set_title_and_labels(axes[3], 'spectrum', 'frequency', ylabel)
+        axes[3].yaxis.labelpad = 0
+        axes[3].set_xlim(freqs[[0, -1]])
+        ylim = axes[3].get_ylim()
+        air = np.diff(ylim)[0] * 0.1
+        axes[3].set_ylim(ylim[0] - air, ylim[1] + air)
+
+        # epoch variance
+        set_title_and_labels(axes[4], 'epochs variance', 'epoch', 'AU')
+
+        if more_fig:
```

Comment:
I had this (maybe weird) idea to return the figure if only one figure plotted and a list of figures if multiple. But you're right, it's better to have stable return type.